replica/table.cc

```diff
@@ -530,6 +530,16 @@ void compaction_group::backlog_tracker_adjust_charges(const std::vector<sstables
     tracker.replace_sstables(old_sstables, new_sstables);
 }
 
+void compaction_group::update_max_sstable_timestamp() {
+    _max_sstable_timestamp = api::missing_timestamp;
+    _main_sstables->for_each_sstable([this] (const sstables::shared_sstable& sst) {
+        _max_sstable_timestamp = std::max(_max_sstable_timestamp, sst->get_stats_metadata().max_timestamp);
+    });
+    _maintenance_sstables->for_each_sstable([this] (const sstables::shared_sstable& sst) {
+        _max_sstable_timestamp = std::max(_max_sstable_timestamp, sst->get_stats_metadata().max_timestamp);
+    });
```

Comment:
we can pass const sstables::sstable_set& here, and only iterate through set which had changes. I initially thought it was enough to track max sstable timestamp (seen), since it's monotonically increasing, but it's better to not rely on assumption that newer sstables will only be added through do_add_sstable(). it might change later.

Reply:
ah I understand you iterate through both sets, so it can track the actual max timestamp. not the max timestamp seen, but I guess the latter is fine as long as it's documented?

Reply:
maybe we can extend sstable_set to provide max_timestamp (currently it provides bytes on disk stat, see: https://github.com/scylladb/scylladb/commit/f54ab4827334373dfa203a0285fb090d212654b5.

```
sstables_set_stats {
    uint64_t bytes_on_disk;
    api::timestamp max_timestamp;
    
    sstables_set_stats& operator+=(const sstables_set_stats& other) noexcept;
}

then you change 

sstable_set_impl:
uint64_t add_bytes_on_disk(uint64_t delta) noexcept;

to become
sstable_set_stats update_stats(const shared_sstable&) noexcept;

sstable_set_impl:
virtual uint64_t bytes_on_disk() const noexcept

becomes
virtual sstable_set_stats stats() const noexcept
```

Reply:
I have updated the compaction_group to track the maximum seen timestamp as discussed.